### PR TITLE
fix: Fix lost extension after renaming a document - EXO-60015

### PR DESCRIPTION
**Pull request by sofyenne**
Prior this change, when we upload a file on eXo 6.3.0 and rename it, the extension is lost even if we migrate to last eXo 6.3.x or 6.4.x versions. This is caused by the clean name method which replaces the `. ` character by `-`.
After this commit:
- For the first case, we check the files uploaded and not renamed on eXo 6.3.0 and recover the extension from the property exo:title when renaming this file. 
- For the second case, we check the files uploaded and renamed on eXo 6.3.0 and recover the extension, based on the file mimetype and an NtFilesExtension.json file which contains the associations between well known mimeTypes and extensions.

### CODE REVIEW

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java [hunk 1]
```diff
@@ -586,9 +586,9 @@ public static long getUserIdentityId(IdentityManager identityManager, String use
    */
   public static String cleanName(String oldName) {
     if (org.apache.commons.lang.StringUtils.isEmpty(oldName)) return oldName;
-    String extention ="" ;
+    String extension ="" ;
```
Comment (azayati) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/util/JCRDocumentsUtil.java]:
```suggestion
    String extension = "" ;
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 2]
```diff
@@ -263,7 +265,8 @@ export default {
                 id: id,
                 repository: 'repository',
                 workspace: 'collaboration',
-                title: decodeURI(attachment.title),
+                //concat the file type if attachement title haven't extension on preview mode
+                title: decodeURI(attachment.title).lastIndexOf('.') >-1 ?decodeURI(attachment.title):decodeURI(attachment.title).concat(this.fileType),
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]:
```suggestion
                title: decodeURI(attachment.title).lastIndexOf('.') >= 0 ? decodeURI(attachment.title) : decodeURI(attachment.title).concat(this.fileType),
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 3]
```diff
@@ -296,4 +299,4 @@ export default {
     }
   }
 };
-</script>
+</script>>
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]:
```suggestion
</script>
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 4]
```diff
@@ -207,7 +208,8 @@ export default {
       return this.file.id===this.fileToEditId;
     },
     fileType() {
-      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')):'';
+      //get extension from the filetypeextension if file name haven't extention 
+      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')): jsonFileExtension[this.file.mimeType] ||'' ;
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]:
```suggestion
      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')) : jsonFileExtension[this.file.mimeType] || '' ;
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 5]
```diff
@@ -101,6 +101,7 @@
   </div>
 </template>
 <script>
+import jsonFileExtension from '../../../../../../../resources/locale/NtFileExtension.json';
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]: IMO the json file should not be added into locales, you can put it directly into the parent folder or at least inti a json folder under [vue-app/documents/](https://github.com/exoplatform/documents/tree/develop/documents-webapp/src/main/webapp/vue-app)

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue [hunk 6]
```diff
@@ -94,7 +94,8 @@ export default {
     renameFile(newTitle){
       this.$root.$emit('cancel-edit-mode', this.file);
       this.file.name = this.fileName.concat(this.fileType);
-      this.$root.$emit('documents-rename', this.file,newTitle);
+      //concat the file type to the new tilte when renaming file
+      this.$root.$emit('documents-rename', this.file,newTitle.concat(this.fileType));
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue]:
```suggestion
      this.$root.$emit('documents-rename', this.file, newTitle.concat(this.fileType));
```

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java [hunk 7]
```diff
@@ -631,15 +631,16 @@ public void renameDocument(long ownerId, String documentID, String title, Identi
       }
       String name = Text.escapeIllegalJcrChars(cleanName(title));
       //clean node name
-
       name = URLDecoder.decode(name,"UTF-8");
-
-      String oldName = node.getName();
-      if (oldName.indexOf('.') != -1 && node.isNodeType(NodeTypeConstants.NT_FILE)) {
+      if(name.indexOf('.') == -1){
```
Comment (azayati) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]:
```suggestion
      if (name.indexOf('.') == -1) {
```

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java [hunk 8]
```diff
@@ -631,15 +631,16 @@ public void renameDocument(long ownerId, String documentID, String title, Identi
...
+      String oldName = node.getName().indexOf('.') == -1 && node.isNodeType(NodeTypeConstants.NT_FILE)
+                           && node.hasProperty(NodeTypeConstants.EXO_TITLE) ? node.getProperty(NodeTypeConstants.EXO_TITLE).getString(): node.getName();
+       if (oldName.indexOf('.') != -1 && node.isNodeType(NodeTypeConstants.NT_FILE)) {
```
Comment (azayati) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: fix indentation

**Review by azayati — CHANGES_REQUESTED** [review]
see review

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue (hunk 4 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]: done !

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue (hunk 5 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]: done !

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue (hunk 6 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue]: done !

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java (hunk 8 above)
Comment (sofyenne) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: done !

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 9]
```diff
@@ -207,7 +208,8 @@ export default {
       return this.file.id===this.fileToEditId;
     },
     fileType() {
-      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')):'';
+      //get extension from the filetypeextension if file name haven't extention 
+      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')) : jsonFileExtension[this.file.mimeType] || '' ;
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]:
```suggestion
      return this.file.name.lastIndexOf('.') >= 0 && !this.file.folder ? this.file.name.substring(this.file.name.lastIndexOf('.')) : ntFileExtension[this.file.mimeType] || '' ;
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue [hunk 10]
```diff
@@ -101,6 +101,7 @@
   </div>
 </template>
 <script>
+import jsonFileExtension from '../../../../json/NtFileExtension.json';
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]:
```suggestion
import ntFileExtension from '../../../../json/NtFileExtension.json';
```

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue [hunk 11]
```diff
@@ -94,8 +94,9 @@ export default {
     renameFile(newTitle){
       this.$root.$emit('cancel-edit-mode', this.file);
       this.file.name = this.fileName.concat(this.fileType);
-      this.$root.$emit('documents-rename', this.file,newTitle);
+      //concat the file type to the new tilte when renaming file
+      this.$root.$emit('documents-rename', this.file , newTitle.concat(this.fileType));
```
Comment (azayati) [inline documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue]:
```suggestion
      this.$root.$emit('documents-rename', this.file, newTitle.concat(this.fileType));
```

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java (hunk 8 above)
Comment (azayati) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: indentation problem is not fixed

**Review by azayati — CHANGES_REQUESTED** [review]
you need to change the source branch on feature/maintenance not develop

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java (hunk 8 above)
Comment (sofyenne) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: fixed

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue (hunk 9 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]: done !

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue (hunk 10 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileNameCell.vue]: done !

File: documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue (hunk 11 above)
Comment (sofyenne) [reply documents-webapp/src/main/webapp/vue-app/documents/components/body/table/cells/DocumentsFileEditNameCell.vue]: done !

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java (hunk 8 above)
Comment (azayati) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: https://github.com/exoplatform/documents/blob/d56e663a4f7de2430b4899aadd0ae3fd97b49372/documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java#L579-L583

still not good

**Review by azayati — DISMISSED** [review]
LGTM

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**